### PR TITLE
Fix and test of compatibility with python 3.7, 3.9 and 3.10

### LINKED ISSUE #479: Error reported when import tensorly (version: 0.8.0) in Python 3.7
In Pyhone 3.7, importing TensorLy, `import tensorly as tl`, will get the following error message:

```
from .generalised_inner_product import inner
  File "<fstring>", line 1
    (tensor1.shape=)
                  ^
SyntaxError: invalid syntax
```

The problem should be caused by the updates in the following lines

https://github.com/tensorly/tensorly/blob/8ff7e2950ca939c64641bfe0169e455f7cc59b56/tensorly/tenalg/core_tenalg/generalised_inner_product.py#L34

https://github.com/tensorly/tensorly/blob/8ff7e2950ca939c64641bfe0169e455f7cc59b56/tensorly/tenalg/core_tenalg/generalised_inner_product.py#L48

which only work in Python 3.8 and 3.9. Will TensorLy stop supporting Python 3.7 and 3.6?

### PR DESCRIPTION
**Pull request by aarmey**
Fixes #479 and #473.

### CODE REVIEW

File: tensorly/utils/tests/test_prod.py [hunk 1]
```diff
@@ -1,10 +0,0 @@
-from .._prod import _prod
-from math import prod
-import tensorly as tl
```
Comment (aarmey) [inline tensorly/utils/tests/test_prod.py:3]: I figured, balancing the tradeoffs, that it was better to remove this test but allow testing on Python 3.7. This fallback rarely occurs, has previously been tested, and will be removed soon when 3.7 reaches end-of-life.

Comment (JeanKossaifi) [reply tensorly/utils/tests/test_prod.py:3]: Makes sense -- can we instead just comment it out (or better yet, skip for Python >= 3.8.x) and make a note to remove utils.prod altogether when we drop Python 3.7 support?

**Comment by JeanKossaifi** [comment]
Aside from the small comment on that test, looks great to me, thanks @aarmey !

**Review by JeanKossaifi — APPROVED** [review]

**Comment by aarmey** [comment]
Thanks @JeanKossaifi. Added back the test based on Python version and am merging!

**Pull request merged**